app/code/Magento/Sales/Cron/CleanExpiredQuotes.php

```diff
@@ -85,6 +85,6 @@ protected function getExpireQuotesAdditionalFilterFields()
      */
     public function setExpireQuotesAdditionalFilterFields(array $fields)
```

Comment:
Hi @sankalpshekhar. Thanks for collaboration. The `setExpireQuotesAdditionalFilterFields` sounds like this method should override value. In your case you case the method should be named something like `addExpireQuotesAdditionalFilterFields`, but according to our backward compatibility guide we can't introduce new public methods.

Can you please provide use case for this improvements?

Reply:
Hello @VladimirZaets 

Currently the method `setExpireQuotesAdditionalFilterFields()` is used in `Magento\PersistentHistory\Plugin\CleanExpiredQuotesPlugin` in _Commerce_ edition of Magento to specify additional condition which in turn is added to filter the quotes collection for deletion. The value passed to the method, overrides the one class attribute may already poses.

For e.g. If I have to add another condition to the filter collection, I can't without losing the previous set condition. If I add a new plugin and try to add a new condition, I either have to repeat the conditions previously added somewhere else in my module, or loose them as my plugin would override the previous values, depending on the sort order of the plugins being loaded.

The 2 possible solutions to remove this impediment (keeping backward compatibility in sight) are as follows:

1. Since `$expireQuotesFilterFields` is defined as an array, we could use the array_merge as I suggested in PR,
2. change `getExpireQuotesAdditionalFilterFields` to a _public_ method, so I could use this to retrieve & manipulate the array values within my plugin, but users could also end up creating plugins on top of this method.

Reply:
@VladimirZaets  Let me know, if additional information would be required here.

Reply:
any updates here? is there anything that i could help out with?